### ISSUE TITLE
Build fails 

### ISSUE THREAD

**Issue by IgrauBezRuk** [Microsoft/glTF-DXViewer#2]
Hey! Thanks for example, I really wanna get deep into it, but after clone I have this output

```
Severity	Code	Description	Project	File	Line	Suppression State
Warning	LNK4099	PDB 'GLTFSDK.pdb' was not found with 'GLTFSDK.lib(Math.obj)' or at 'C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\x64\Debug\GLTFSDK.pdb'; linking object as if no debug info	gltfparser	C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\gltfparser\GLTFSDK.lib(Math.obj)	1	
Warning	C4267	'=': conversion from 'size_t' to 'unsigned int', possible loss of data	gltfparser	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\sharedgltfparser\glbgltfconverter.cpp	201	
Warning	C4267	'=': conversion from 'size_t' to 'unsigned int', possible loss of data	gltfparser	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\sharedgltfparser\glbgltfconverter.cpp	264	
Warning	C4267	'=': conversion from 'size_t' to 'unsigned int', possible loss of data	gltfparser	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\sharedgltfparser\glbgltfconverter.cpp	266	
Warning	C4267	'=': conversion from 'size_t' to 'unsigned int', possible loss of data	gltfparser	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\sharedgltfparser\glbgltfconverter.cpp	268	
Warning	C4267	'=': conversion from 'size_t' to 'unsigned int', possible loss of data	WinRTGLTFParser	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\sharedgltfparser\glbgltfconverter.cpp	201	
Warning	C4267	'=': conversion from 'size_t' to 'unsigned int', possible loss of data	WinRTGLTFParser	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\sharedgltfparser\glbgltfconverter.cpp	264	
Warning	C4267	'=': conversion from 'size_t' to 'unsigned int', possible loss of data	WinRTGLTFParser	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\sharedgltfparser\glbgltfconverter.cpp	266	
Warning	C4267	'=': conversion from 'size_t' to 'unsigned int', possible loss of data	WinRTGLTFParser	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\sharedgltfparser\glbgltfconverter.cpp	268	
Warning	LNK4099	PDB 'GLTFSDK.pdb' was not found with 'GLTFSDK.lib(Color.obj)' or at 'C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\x64\Debug\GLTFSDK.pdb'; linking object as if no debug info	gltfparser	C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\gltfparser\GLTFSDK.lib(Color.obj)	1	
Warning	LNK4099	PDB 'GLTFSDK.pdb' was not found with 'GLTFSDK.lib(Color.obj)' or at 'C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\x64\Debug\WinRTGLTFParser\GLTFSDK.pdb'; linking object as if no debug info	WinRTGLTFParser	C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\WinRTGLTFParser\GLTFSDK.lib(Color.obj)	1	
Warning	LNK4099	PDB 'GLTFSDK.pdb' was not found with 'GLTFSDK.lib(Deserialize.obj)' or at 'C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\x64\Debug\GLTFSDK.pdb'; linking object as if no debug info	gltfparser	C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\gltfparser\GLTFSDK.lib(Deserialize.obj)	1	
Warning	LNK4099	PDB 'GLTFSDK.pdb' was not found with 'GLTFSDK.lib(Deserialize.obj)' or at 'C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\x64\Debug\WinRTGLTFParser\GLTFSDK.pdb'; linking object as if no debug info	WinRTGLTFParser	C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\WinRTGLTFParser\GLTFSDK.lib(Deserialize.obj)	1	
Warning	LNK4099	PDB 'GLTFSDK.pdb' was not found with 'GLTFSDK.lib(GLBResourceReader.obj)' or at 'C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\x64\Debug\GLTFSDK.pdb'; linking object as if no debug info	gltfparser	C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\gltfparser\GLTFSDK.lib(GLBResourceReader.obj)	1	
Warning	LNK4099	PDB 'GLTFSDK.pdb' was not found with 'GLTFSDK.lib(GLBResourceReader.obj)' or at 'C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\x64\Debug\WinRTGLTFParser\GLTFSDK.pdb'; linking object as if no debug info	WinRTGLTFParser	C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\WinRTGLTFParser\GLTFSDK.lib(GLBResourceReader.obj)	1	
Warning	LNK4099	PDB 'GLTFSDK.pdb' was not found with 'GLTFSDK.lib(GLTFDocument.obj)' or at 'C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\x64\Debug\GLTFSDK.pdb'; linking object as if no debug info	gltfparser	C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\gltfparser\GLTFSDK.lib(GLTFDocument.obj)	1	
Warning	LNK4099	PDB 'GLTFSDK.pdb' was not found with 'GLTFSDK.lib(GLTFDocument.obj)' or at 'C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\x64\Debug\WinRTGLTFParser\GLTFSDK.pdb'; linking object as if no debug info	WinRTGLTFParser	C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\WinRTGLTFParser\GLTFSDK.lib(GLTFDocument.obj)	1	
Warning	LNK4099	PDB 'GLTFSDK.pdb' was not found with 'GLTFSDK.lib(Math.obj)' or at 'C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\x64\Debug\WinRTGLTFParser\GLTFSDK.pdb'; linking object as if no debug info	WinRTGLTFParser	C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\WinRTGLTFParser\GLTFSDK.lib(Math.obj)	1	
Warning	LNK4099	PDB 'GLTFSDK.pdb' was not found with 'GLTFSDK.lib(Schema.obj)' or at 'C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\x64\Debug\GLTFSDK.pdb'; linking object as if no debug info	gltfparser	C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\gltfparser\GLTFSDK.lib(Schema.obj)	1	
Warning	LNK4099	PDB 'GLTFSDK.pdb' was not found with 'GLTFSDK.lib(Schema.obj)' or at 'C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\x64\Debug\WinRTGLTFParser\GLTFSDK.pdb'; linking object as if no debug info	WinRTGLTFParser	C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\WinRTGLTFParser\GLTFSDK.lib(Schema.obj)	1	
Warning	LNK4099	PDB 'GLTFSDK.pdb' was not found with 'GLTFSDK.lib(Serialize.obj)' or at 'C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\x64\Debug\GLTFSDK.pdb'; linking object as if no debug info	gltfparser	C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\gltfparser\GLTFSDK.lib(Serialize.obj)	1	
Warning	LNK4099	PDB 'GLTFSDK.pdb' was not found with 'GLTFSDK.lib(Serialize.obj)' or at 'C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\x64\Debug\WinRTGLTFParser\GLTFSDK.pdb'; linking object as if no debug info	WinRTGLTFParser	C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\WinRTGLTFParser\GLTFSDK.lib(Serialize.obj)	1	
Warning	LNK4099	PDB 'GLTFSDK.pdb' was not found with 'GLTFSDK.lib(Validation.obj)' or at 'C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\x64\Debug\GLTFSDK.pdb'; linking object as if no debug info	gltfparser	C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\gltfparser\GLTFSDK.lib(Validation.obj)	1	
Warning	LNK4099	PDB 'GLTFSDK.pdb' was not found with 'GLTFSDK.lib(Validation.obj)' or at 'C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\x64\Debug\WinRTGLTFParser\GLTFSDK.pdb'; linking object as if no debug info	WinRTGLTFParser	C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\WinRTGLTFParser\GLTFSDK.lib(Validation.obj)	1	
Error	X4502	invalid vs_2_0 output semantic 'SV_TARGET'	ModelViewer	C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\ModelViewer\Content\SimplePixelShader.hlsl	9	
```

Seems like `SimplePixelShader.hlsl` has wrong type and version, because its pixel shader.
If I set in this shader setting type `PixelShader`, I have next errors.

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	81	
Warning	C4267	'=': conversion from 'size_t' to 'int', possible loss of data	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\buffercache.cpp	9	
Error	D8040	error creating or communicating with child process	ModelViewer	C:\Users\nkrupitskas\Documents\IgrauBezRuk\glTF-DXViewer\ModelViewer\cl	1	
Warning	C4267	'return': conversion from 'size_t' to 'int', possible loss of data	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\scene\graphcontainernode.cpp	142	
Warning	C4267	'initializing': conversion from 'size_t' to 'int', possible loss of data	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\scene\meshnode.cpp	87	
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\modelfactory.cpp	107	
Error	C3312	no callable 'await_resume' function found for type 'std::future<void>'	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\modelfactory.cpp	107	
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\modelfactory.cpp	119	
Warning	C4267	'return': conversion from 'size_t' to 'unsigned int', possible loss of data	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\nodematerial.h	57	
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\viewmodels\rootpageviewmodel.cpp	29	
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\viewmodels\rootpageviewmodel.cpp	48	
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\viewmodels\rootpageviewmodel.cpp	50	
Error	C2440	'return': cannot convert from 'nullptr' to 'std::future<std::shared_ptr<GraphNode>>'	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\viewmodels\rootpageviewmodel.cpp	50	
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\viewmodels\rootpageviewmodel.cpp	58	
Error	C3312	no callable 'await_resume' function found for type 'std::future<std::shared_ptr<GraphNode>>'	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\viewmodels\rootpageviewmodel.cpp	58	
Error	C1903	unable to recover from previous error(s); stopping compilation	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\viewmodels\rootpageviewmodel.cpp	58	
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	51	
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	52	
Error	C2440	'return': cannot convert from 'Windows::Storage::Streams::IBuffer ^' to 'std::future<Windows::Storage::Streams::IBuffer ^>'	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	53	
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	67	
Error	C2440	'return': cannot convert from 'void *' to 'std::future<void *>'	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	67	
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	75	
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	77	
Error	C2440	'return': cannot convert from 'std::vector<byte,std::allocator<_Ty>>' to 'std::future<std::vector<byte,std::allocator<_Ty>>>'	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	81	
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	128	
Error	C2679	binary '=': no operator found which takes a right-hand operand of type 'std::future<std::vector<byte,std::allocator<_Ty>>> (__cdecl *)(Windows::Storage::StorageFolder ^,Platform::String ^,Platform::String ^,int,uint32_t &,uint32_t &)' (or there is no acceptable conversion)	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	129	
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	170	
Error	C2440	'return': cannot convert from 'ModelViewer::Sample3DSceneRenderer::TexWrapper' to 'std::future<ModelViewer::Sample3DSceneRenderer::TexWrapper>'	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	170	
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	465	
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	466	
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	519	
Error	C2440	'return': cannot convert from 'ModelViewer::Sample3DSceneRenderer::TexWrapper' to 'std::future<ModelViewer::Sample3DSceneRenderer::TexWrapper>'	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	519	
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	527	
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	528	
Error	C3536	'res': cannot be used before it is initialized	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	530	
Error	C2039	'texResourceView': is not a member of 'default::int32'	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	530	
Error	C2039	'texSampler': is not a member of 'default::int32'	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	531	
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	534	
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	535	
Error	C2039	'texResourceView': is not a member of 'default::int32'	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	537	
Error	C2039	'texSampler': is not a member of 'default::int32'	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	538	
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	542	
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	546	
Error	C2039	'texResourceView': is not a member of 'default::int32'	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	548	
Error	C2039	'texSampler': is not a member of 'default::int32'	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	549	
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	551	
Error	C2561	'ModelViewer::Sample3DSceneRenderer::CreateEnvironmentMapResourcesAsync': function must return a value	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	551	
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	641	
Error	C3773	please use /await compiler switch to enable coroutines	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	648	
Error	C2561	'ModelViewer::Sample3DSceneRenderer::CreateDeviceDependentResources': function must return a value	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\content\sample3dscenerenderer.cpp	648	
Warning	C4267	'=': conversion from 'size_t' to 'int', possible loss of data	ModelViewer	c:\users\nkrupitskas\documents\igraubezruk\gltf-dxviewer\modelviewer\shadercache.cpp	9	

```
After adding /await to compiler command line, it's successfully built.

Is it expected or why is it happening?

**Comment by peted70** [comment]
Thanks for flagging it - I suspect you are building for x64 and I think some of the settings for that haven't been updated correctly. I will fix this but In the meantime you can build for x86.

**Comment by peted70** [comment]
There have been fixes for the 64bit build in this commit 6c784cbe380cd0ce303a4d6b99ca7fa066fba432 this includes fixing all of the size_t warnings and the various compiler flags and settings.

**Comment by IgrauBezRuk** [comment]
Yes, now it working! Thanks. :)
